lib/fs.js

```diff
@@ -1164,8 +1164,8 @@ function toUnixTimestamp(time) {
   if (typeof time === 'string' && +time == time) {
     return +time;
   }
-  if (typeof time === 'number') {
-    if (!Number.isFinite(time) || time < 0) {
+  if (typeof time === 'number' && !isNaN(time) && isFinite(time)) {
```

Comment:
I think `Number.isFinite(time)` would cover `typeof time === 'number'` anyway, so this is equivalent to just `Number.isFinite(time)`.